src/Parser.php

```diff
@@ -57,13 +58,38 @@ private function parseModel(string $className, ParserContext $context, RawClassM
         }
         $registry->add($rawClassMetadata);
 
+        $this->parseDiscriminatorClasses($rawClassMetadata, $registry);
+
         foreach ($rawClassMetadata->getPropertyVariations() as $property) {
             $type = $property->getType();
-            if ($type instanceof PropertyTypeIterable) {
-                $type = $type->getLeafType();
-            }
-            if ($type instanceof PropertyTypeClass) {
-                $this->parseModel($type->getClassName(), $context->push($property), $registry);
```

Comment:
from the context class, it looks like it was made to provide the reference to where the error happened in case something does not work. but it is completely unused, so i agree we can drop it.